File: Round1/B/B.py
from itertools import *

def gen(s):
    return [len(list(gp)) for _, gp in groupby(s)]

cas = int(input())
for casi in range(cas):
    n = int(input())
    a, b = input(), input()
    a = ''.join([b[0], a, b[-1]])
    b = ''.join([b[0], b, b[-1]])
    a, b = gen(a), gen(b)

    i = 0
    ans = 0
    for t in b:
        j = i
        while j < len(a) and t > 0:
            t -= a[j]
            j += 1
        if t < 0 or (j - i) % 2 == 0:
            ans = -1
            break
        ans += (j - i) // 2
        i = j

    print(ans)

File: Round1/B/B.go
package main

import (
    "bufio"
    "fmt"
    "os"
    "strconv"
    "strings"
)

func ReadLine(reader *bufio.Reader) string {
    line, _ := reader.ReadString('\n')
    return strings.TrimRight(line, "\n")
}

func ReadInt(reader *bufio.Reader) int {
    num, _ := strconv.Atoi(ReadLine(reader))
    return num
}

func Gen(a string) []int {
    n := len(a)
    var ret []int
    ret = append(ret, 1)
    for i := 1; i < n; i++ {
        if a[i] != a[i - 1] {
            ret = append(ret, 1)
        } else {
            ret[len(ret) - 1] += 1
        }
    }
    return ret
}

func main() {
    reader := bufio.NewReader(os.Stdin)
    cas := ReadInt(reader)
    for casi := int(0); casi < cas; casi++ {
        n := ReadInt(reader)
        ra := ReadLine(reader)
        rb := ReadLine(reader)

        a := Gen(rb[: 1] + ra + rb[n - 1: ])
        b := Gen(rb[: 1] + rb + rb[n - 1: ])

        i, ans := 0, 0
        for _, t := range b {
            j := i
            for ; j < len(a) && t > 0; j++ {
                t -= a[j]
            }
            if (j - i) % 2 == 0 || t < 0 {
                ans = -1
                break
            }
            ans += (j - i) / 2
            i = j
        }
        fmt.Println(ans)
    }
}
